prog/dftb+/lib_dftb/populations.F90

```diff
@@ -451,5 +451,64 @@ subroutine getChargePerShell(qOrb, orb, species, chargePerShell, qRef)
   end subroutine getChargePerShell
 
 
+  !> Block mulliken analysis with dense lower triangle matrices.
+  subroutine denseBlockMulliken(rhoSqr, overSqr, iSquare, qq)
+
+    !> Square (lower triangular) spin polarized density matrix
+    real(dp), intent(in) :: rhoSqr(:,:,:)
+
+    !> Square (lower triangular) overlap matrix
+    real(dp), intent(in) :: overSqr(:,:)
+
+    !> Atom positions in the row/column of square matrices
+    integer, intent(in) :: iSquare(:)
+
+    !> Mulliken block charges on output (mOrb, mOrb, nAtom, nSpin)
+    real(dp), intent(out) :: qq(:,:,:,:)
+
+    real(dp), allocatable :: tmpS(:,:)
+    real(dp), allocatable :: tmpD(:,:)
+
+    integer :: nAOs, nat, nSpin, ii, jj, kk, iat, iS, nOrb
+
+    nSpin = size(rhoSqr, dim=3)
+    nat = size(iSquare, dim=1) - 1
+    nAOs = size(rhoSqr, dim=1)
+
+    allocate(tmpS(nAOs,nAOs))
+    allocate(tmpD(nAOs,nAOs))
+
+    ! ... symmetrize overlap
```

Comment:
Please remove the `...` from the comments, as their are neither informative nor improve readability